specviz/plugins/unit_change/tests/test_unit_conversion_gui.py

```diff
@@ -1,59 +1,137 @@
-import pytest
-
 from qtpy import QtCore
-from qtpy.QtWidgets import QMainWindow, QMdiSubWindow, QListWidget, QAction, QDialog, QDialogButtonBox
-
-from ..unit_change_dialog import UnitChangeDialog
-
-
-# def test_custom_units_correct(qtbot):
-#     uc = UnitChangeDialog()
-#     uc.show()
-#     qtbot.addWidget(uc)
-#
-#     uc.ui.comboBox_spectral.setCurrentIndex(2)
-#     assert uc.ui.comboBox_spectral.currentText() == uc._units_titles[2]
-#
-#     uc.ui.comboBox_spectral.setCurrentIndex(uc.ui.comboBox_spectral.count()-1)
-#     assert uc.ui.comboBox_spectral.currentText() == "Custom"
-#
-#     uc.ui.line_custom_spectral.setText("fT")
-#     assert uc.ui.on_accepted() == True
-#
-#
-# def test_custom_units_incorrect(qtbot):
-#     uc = UnitChangeDialog()
-#     uc.show()
-#     qtbot.addWidget(uc)
-#
-#     uc.ui.comboBox_spectral.setCurrentIndex(uc.ui.comboBox_spectral.count() - 1)
-#     assert uc.ui.comboBox_spectral.currentText() == "Custom"
-#
-#     uc.ui.line_custom_spectral.setText("feet")
-#     assert uc.ui.on_accepted() == False
-#
-#
-# def test_accept_works_correctly(qtbot):
-#     uc = UnitChangeDialog()
-#     uc.show()
-#     qtbot.addWidget(uc)
-#
-#     uc.ui.comboBox_spectral.setCurrentIndex(4)
-#     assert uc.ui.comboBox_spectral.currentText() == uc._units_titles[4]
-#
-#     qtbot.mouseClick(uc.ui.buttonBox.button(QDialogButtonBox.Ok), QtCore.Qt.LeftButton)
-#     assert uc.current_units == uc._units_titles[4]
-#
-#
-# def test_cancel_works_correctly(qtbot):
-#     uc = UnitChangeDialog()
-#     uc.show()
-#     qtbot.addWidget(uc)
-#
-#     old_units = uc.current_units
-#
-#     uc.ui.comboBox_spectral.setCurrentIndex(3)
-#     assert uc.ui.comboBox_spectral.currentText() == uc._units_titles[3]
-#
-#     qtbot.mouseClick(uc.ui.buttonBox.button(QDialogButtonBox.Cancel), QtCore.Qt.LeftButton)
-#     assert uc.current_units == old_units
+from qtpy.QtWidgets import QDialogButtonBox
+
+from specviz.plugins.unit_change.unit_change_dialog import UnitChangeDialog
+
+
+def get_ucd(specviz_gui):
+    ucd = UnitChangeDialog(specviz_gui.current_workspace)
+    ucd.show()
+    return ucd
```

Comment:
@javerbukh I've made yet another minor change to this in #544, so ultimately I think this function will end up looking like the following:

```python
def get_ucd(specviz_gui):
    ucd = specviz_gui.current_workspace._plugins['Unit Change Dialog']
    ucd.show()
    return ucd
```

This will have the advantage of not adding another icon to the workspace toolbar. I don't think this needs to be changed before merging, though.